components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/claims/BFSIDefaultOIDCClaimsCallbackHandler.java

```diff
@@ -0,0 +1,169 @@
+                if (jwtClaimsSet != null) {
+                    userClaimsInOIDCDialect.putAll(jwtClaimsSet.getClaims());
+                }
+                addCnfClaimToOIDCDialect(tokenReqMessageContext, userClaimsInOIDCDialect);
+                addConsentIDClaimToOIDCDialect(tokenReqMessageContext, userClaimsInOIDCDialect);
+                updateSubClaim(tokenReqMessageContext, userClaimsInOIDCDialect);
+
+                for (Map.Entry<String, Object> claimEntry : userClaimsInOIDCDialect.entrySet()) {
+                    if (IdentityCommonConstants.SCOPE.equals(claimEntry.getKey())) {
+                        String[] nonInternalScopes = IdentityCommonUtils
+                                .removeInternalScopes(claimEntry.getValue().toString()
+                                        .split(IdentityCommonConstants.SPACE_SEPARATOR));
+                        jwtClaimsSetBuilder.claim(IdentityCommonConstants.SCOPE, StringUtils.join(nonInternalScopes,
+                                IdentityCommonConstants.SPACE_SEPARATOR));
+                    } else {
+                        jwtClaimsSetBuilder.claim(claimEntry.getKey(), claimEntry.getValue());
+                    }
+                }
+                return jwtClaimsSetBuilder.build();
+            }
+        } catch (RequestObjectException e) {
+            throw new IdentityOAuth2Exception(e.getMessage(), e);
+        }
+        return super.handleCustomClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
+    }
+
+    @Generated(message = "Excluding from code coverage since it makes is used to return claims from the super class")
+    public JWTClaimsSet getJwtClaimsFromSuperClass(JWTClaimsSet.Builder jwtClaimsSetBuilder,
+                                                   OAuthTokenReqMessageContext tokenReqMessageContext)
+            throws IdentityOAuth2Exception {
+
+        return super.handleCustomClaims(jwtClaimsSetBuilder, tokenReqMessageContext);
+    }
+
+    private void addCnfClaimToOIDCDialect(OAuthTokenReqMessageContext tokenReqMessageContext,
+                                          Map<String, Object> userClaimsInOIDCDialect) {
+        Base64URL certThumbprint;
+        X509Certificate certificate;
+        String headerName = IdentityCommonUtils.getMTLSAuthHeader();
+
+        HttpRequestHeader[] requestHeaders = tokenReqMessageContext.getOauth2AccessTokenReqDTO()
+                .getHttpRequestHeaders();
+        Optional<HttpRequestHeader> certHeader =
+                Arrays.stream(requestHeaders).filter(h -> headerName.equals(h.getName())).findFirst();
+        if (certHeader.isPresent()) {
+            try {
+                certificate = IdentityCommonUtils.parseCertificate(certHeader.get().getValue()[0]);
+                certThumbprint = X509CertUtils.computeSHA256Thumbprint(certificate);
+                userClaimsInOIDCDialect.put("cnf", Collections.singletonMap("x5t#S256", certThumbprint));
```

Comment:
Let's add these string values as constants in ```IdentityCommonConstants.java```.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136